lib/pageflow/policies/folder_policy.rb

```diff
@@ -0,0 +1,51 @@
+module Pageflow
+  module Policies
+    class FolderPolicy
+      class Scope
+        attr_reader :user, :scope
+
+        def initialize(user, scope)
+          @user = user
+          @scope = scope
+        end
+
+        def resolve
+          scope.where('account_id IN (?) OR id IN (?)',
+                      accounts_where_user_is_at_least_publisher(user).map(&:id),
+                      user.entries.map(&:folder_id))
+        end
+
+        private
+
+        def sanitize_sql_array(array)
+          ActiveRecord::Base.send(:sanitize_sql_array, array)
+        end
+
+        def accounts_where_user_is_at_least_publisher(user)
+          user.membership_accounts.joins(sanitize_sql_array([
+            'LEFT OUTER JOIN pageflow_memberships as pageflow_memberships_2 ON ' \
+            'pageflow_memberships_2.user_id = :user_id AND ' \
+            'pageflow_memberships_2.entity_type = "Pageflow::Account" AND ' \
+            'pageflow_memberships_2.entity_id = pageflow_accounts.id AND ' \
+            'pageflow_memberships_2.role IN ("publisher", "manager")',
+            user_id: user.id])).where('pageflow_memberships_2.entity_id IS NOT NULL')
```

Comment:
Because the user gets their accounts via pageflow_memberships also, requiring a way to distinguish the columns